cdap-watchdog/src/main/java/io/cdap/cdap/metrics/process/MessagingMetricsProcessorManagerService.java

```diff
@@ -74,6 +79,10 @@
     metricsWriter.initialize(metricsContext);
     this.metricsWriters.add(metricsWriter);
 
+    for (Map.Entry<String, MetricsWriter> metricsWriterEntry : metricsWriterProvider.loadMetricsWriters().entrySet()) {
+      this.metricsWriters.add(metricsWriterEntry.getValue());
```

Comment:
When the `MetricsWriter.initialize` will be called? I would suggest called them in the startUp method of this service (including the default one, the call should move there).

Reply:
done

Reply:
fixed.